app/controllers/account_controller.rb

```diff
@@ -269,7 +269,8 @@ def change_password
 
   def auth_source_sso_failed
     failure = session.delete :auth_source_sso_failure
-    user = failure[:user]
+    login = failure[:login]
+    user = find_user_from_auth_source(login) || create_user_from_auth_source(login)
```

Comment:
@oliverguenther why do we (potentially) create a user if the login failed?

Reply:
We need the user object that contains the errors from saving it beforehand. Before this PR, this user was added to the session in the error case, which resulted in an internal error trying to serialize a non-saved user. So this exact code has run before the redirect and would be expected to fail. I can rewrite it to _build_ the user instead of trying to save it, even though the result will be the same (in the current way the code works)